appmonitoring/ts/src/CertificateManager.ts

```diff
@@ -155,22 +155,38 @@ export class CertificateManager {
         }
     }
 
-    public async GetSecretDetails(operationId: string, kubeConfig: k8s.KubeConfig): Promise<WebhookCertData> {
+    public async GetSecretDetails(operationId: string, kubeConfig: k8s.KubeConfig, clusterArmId: string, clusterArmRegion: string): Promise<WebhookCertData> {
         try {
             const k8sApi = kubeConfig.makeApiClient(k8s.CoreV1Api);
             const secretsObj: k8s.V1Secret = await k8sApi.readNamespacedSecret({ name: CertificateStoreName, namespace: KubeSystemNamespaceName });
+            let certificate: WebhookCertData = null;
             if (secretsObj.data) {
-                const certificate: WebhookCertData = {
-                    caCert: Buffer.from(secretsObj.data['ca.cert'], 'base64').toString('utf-8'),
-                    caKey: Buffer.from(secretsObj.data['ca.key'], 'base64').toString('utf-8'),
-                    tlsCert: Buffer.from(secretsObj.data['tls.cert'], 'base64').toString('utf-8'),
-                    tlsKey: Buffer.from(secretsObj.data['tls.key'], 'base64').toString('utf-8')
-                };
+                try {
+                    certificate = {
+                        caCert: Buffer.from(secretsObj.data['ca.cert'], 'base64').toString('utf-8'),
+                        caKey: Buffer.from(secretsObj.data['ca.key'], 'base64').toString('utf-8'),
+                        tlsCert: Buffer.from(secretsObj.data['tls.cert'], 'base64').toString('utf-8'),
+                        tlsKey: Buffer.from(secretsObj.data['tls.key'], 'base64').toString('utf-8')
+                    };
+                } catch (error) {
+                    logger.error('Failed to decode secret data from base64!', operationId, this.requestMetadata);
```

Comment:
A single logger.error call please, let's not split a single error message into multiple calls, that creates a false impression for the reader.